lua/neo-tree/sources/common/commands.lua

```diff
@@ -530,9 +530,21 @@ M.show_file_details = function(state)
     table.insert(left, "Size")
     table.insert(right, utils.human_size(stat.size))
     table.insert(left, "Created")
-    table.insert(right, os.date("%Y-%m-%d %I:%M %p", stat.birthtime.sec))
+    local created_format = state.config.created_format
+      or require("neo-tree").config.default_component_configs.created.format
+    if type(created_format) == "function" then
+      table.insert(right, created_format(stat.birthtime.sec))
+    else
+      table.insert(right, os.date(created_format, stat.birthtime.sec))
+    end
```

Comment:
I think you could pull this logic out into a new file like `sources/common/lib/format.lua` just for code reuse purposes:

```lua
-- sources/common/lib/format.lua
local M = {}
M.date = function(format, seconds)
  -- add logic for special case for format = "relative"
  return type(format) == "function" and format(seconds) or os.date(format, seconds)
end
return M 
```

```suggestion
    local date = require('neotree.sources.common.lib.format').date
    local created_format = state.config.created_format
      or require("neo-tree").config.default_component_configs.created.format
    table.insert(right, date(created_format, stat.birthtime.sec))
    local modified_format = state.config.modified_format
      or require("neo-tree").config.default_component_configs.modified.format
    table.insert(right, date(modified_format, stat.mtime.sec))
```

Reply:
It would be cleaner to implement relative timestamps in the utils module as you suggested, then add a commented-out setting in `defaults.lua`.

```lua
-- lua/neo-tree/utils/init.lua
M.relative_date = function(seconds)
  -- Implementation
end
```

```lua
-- lua/neo-tree/defaults.lua
...
last_modified = {
  ...
  --format = require("neo-tree.utils").relative_date
},
...
```

What do you think? I'll implement this along with this PR.

Reply:
sounds good to me!